account_invoice_section_picking/models/account_move_line.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = "account.move.line"
+
+    def _get_section_group(self):
+        """Group invoice lines according to uninvoiced delivery pickings"""
+        group = super()._get_section_group()
+        # If product is invoiced by delivered quantities:
+        #  - filter on done pickings to avoid displaying backorders not yet
+        #    processed
+        #  - Remove pickings linked with same sale order lines if an invoice
+        #    was created after its date_done
+        invoice_section_grouping = self.company_id.invoice_section_grouping
+        if (
+            invoice_section_grouping == "delivery_picking"
+            and self.product_id.invoice_policy == "delivery"
+        ):
+            last_invoice_line = self.search(
+                [
+                    ("sale_line_ids", "in", self.sale_line_ids.ids),
+                    ("state", "!=", "cancel"),
```

Comment:
```suggestion
                    ("parent_state", "!=", "cancel"),
```

as discussed on slack. Tested locally - works fine.